src/cluster/src/check.rs

```diff
@@ -128,3 +169,98 @@ pub fn check_already_installed(helm: &HelmClient, app_repo: &str) -> Result<(),
     }
     Ok(())
 }
+
+/// check if load balancer is up
+pub async fn check_load_balancer_status() -> Result<(), CheckError> {
+    let config = K8Config::load()?;
+    let context = match config {
+        K8Config::Pod(_) => return Ok(()),
+        K8Config::KubeConfig(context) => context,
+    };
+
+    let cluster_context = context
+        .config
+        .current_context()
+        .ok_or(CheckError::NoActiveKubernetesContext)?;
+    let username = cluster_context.context.user.to_owned();
+
+    // create dummy service
+    create_dummy_service()?;
+    if wait_for_service_exist(DEFAULT_NAMESPACE).await?.is_some() {
+        // IP found, everything good
+        delete_service()?;
+    } else {
+        delete_service()?;
+        if username == "minikube" {
+            // incase of macos we need to run tunnel with elevated context of sudo
+            // hence handle both seperately
+            return Err(get_tunnel_error());
+        }
+        return Err(CheckError::LoadBalancerServiceNotAvailable);
+    }
+
+    Ok(())
+}
+
+fn create_dummy_service() -> Result<(), CheckError> {
```

Comment:
What is the dummy service used for? I think I've heard of this before but have never known why we need it.

Reply:
There is no way to check the status of minikube tunnel. So we create a dummy service and check if IP is automaticaly assigned. In this way, if IP is not assigned, we assume the tunnel is down, and if IP is properly assigned we proceed forward.